### PR TITLE
starship: update to 1.10.1

### PR DESCRIPTION
**Pull request by tranzystorek-io**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by tranzystorek-io** [comment]
Starship 1.10 switches its git backend from libgit2 to gitoxide which requires building zlib-ng uncondtionally (for performance optimization reasons).

File: srcpkgs/starship/template [hunk 1]
```diff
@@ -1,19 +1,18 @@
 # Template file for 'starship'
 pkgname=starship
-version=1.9.1
+version=1.10.0
 revision=1
 build_style=cargo
 build_helper=qemu
-hostmakedepends="libgit2-devel pkg-config"
-makedepends="libgit2-devel openssl-devel"
+hostmakedepends="cmake pkg-config"
```
Comment (classabbyamp) [inline srcpkgs/starship/template:7]: what does it need cmake for?

Comment (tranzystorek-io) [reply srcpkgs/starship/template:7]: building zlib-ng

**Comment by icp1994** [comment]
Probably better to for the next patch version https://github.com/starship/starship/issues/4251

**Comment by tranzystorek-io** [comment]
hm, I've been having weird hangs when `cd`ing into the void-packages repo even on the previous version

**Comment by tranzystorek-io** [comment]
This version is ready for merge (1.10.1 includes fix for the bug @icp1994 mentioned)

**Pull request merged**